### PR TITLE
Consistent use of the TopBar and GoBackButton components

### PR DESCRIPTION
**Pull request by QuicheLorraine3600**
This PR aims to bring more consistency within the app: 
- the usage of TopAppBar has been replaced by our custom TopBar in all the app
- screens not yet using our GoBackButton have been updated
- the deprecated version of GoBackButton has been removed
- tests have been updated accordingly

### CODE REVIEW

**Review by juaniac — APPROVED** [review]
Really nice PR, makes all of the top bar's of the app the same.
The functionality of all screens remains the same and the tests all work so this change was nicely merge in.
Thus I approve this PR.

File: app/src/main/java/com/monkeyteam/chimpagne/ui/components/TopBar.kt [hunk 1]
```diff
@@ -26,6 +27,7 @@ fun TopBar(
             text = text,
             style = MaterialTheme.typography.headlineMedium,
             fontFamily = ChimpagneFontFamily,
+            overflow = TextOverflow.Ellipsis,
```
Comment (arnaudlvq) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/components/TopBar.kt:30]: Nice to see some consideration for smaller screens and large text

File: app/src/main/java/com/monkeyteam/chimpagne/ui/event/details/supplies/SuppliesScreen.kt [hunk 2]
```diff
@@ -133,10 +132,9 @@ fun SuppliesScreen(
         }
       },
       topBar = {
-        TopAppBar(
-            title = { Text(stringResource(id = R.string.supplies_screen_title)) },
-            modifier = Modifier.shadow(4.dp),
-            navigationIcon = { GoBackButton(navObject) })
+        TopBar(
```
Comment (arnaudlvq) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/event/details/supplies/SuppliesScreen.kt:135]: Consistency across the app really enhance the user experience

File: app/src/main/java/com/monkeyteam/chimpagne/ui/event/polls/PollAndVotingScreen.kt [hunk 3]
```diff
@@ -97,18 +95,9 @@ fun PollsAndVotingScreen(
...
-                  Modifier.testTag("screen title"))
-            },
-            modifier = Modifier.shadow(4.dp),
-            navigationIcon = {
-              IconButton(onClick = { onGoBack() }) {
-                Icon(Icons.AutoMirrored.Filled.ArrowBack, "back")
-              }
-            })
+        TopBar(
+            text = stringResource(id = R.string.polls_topbar_title),
```
Comment (arnaudlvq) [inline app/src/main/java/com/monkeyteam/chimpagne/ui/event/polls/PollAndVotingScreen.kt:99]: great naming, i see the page, the location on the page and what it displays

**Review by arnaudlvq — APPROVED** [review]
Great, this PR is what we usally think of when we say "small PR". To the point, concise, and well written

**Review by Athanaze — APPROVED** [review]
Good job restoring the proper bar on eventscreen, that somehow got removed.

**Pull request merged**